0.7.0 release breaks data tfe_team sources

Hello -

I was testing some changes and the tfe provider seems to be unhappy with code that works perfectly well under `v0.6.0`. It looks like the request either didn't get data back from the call or the data we did get didn't fulfill the required interface: `panic: interface conversion: interface {} is nil, not map[string]interface {}`.

I've pinned the provider to 0.6.0 to remediate for time time being and the code works.

Terraform/0.11.11 TFE/bf86e8a

Example code:

```hcl
provider "tfe" {
  # XXX version 0.7.0 is downloaded
  version = "~> 0.6"
  token = "${var.tfe_org_token}"
}

data "tfe_team" "pso" {
  name         = "${local.pso_team_name}"
  organization = "${local.tfe_organization}"
}
```

Example error:

```
Error: Error refreshing state: 2 error(s) occurred:

* tfe_workspace.stack: 1 error(s) occurred:

* tfe_workspace.stack: tfe_workspace.stack: unexpected EOF
* data.tfe_team.pso: 1 error(s) occurred:

* data.tfe_team.pso: data.tfe_team.pso: unexpected EOF


panic: interface conversion: interface {} is nil, not map[string]interface {}
2019-02-14T22:22:00.546Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: 
2019-02-14T22:22:00.546Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: goroutine 58 [running]:
2019-02-14T22:22:00.546Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: github.com/terraform-providers/terraform-provider-tfe/tfe.resourceTFEWorkspaceRead(0xc000362a10, 0xdda820, 0xc000001680, 0xc000362a10, 0x0)
2019-02-14T22:22:00.546Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/tfe/resource_tfe_workspace.go:249 +0xf4d
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc00033f880, 0xc0004afd10, 0xdda820, 0xc000001680, 0xc0002dfd28, 0x4c1b01, 0xd29960)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:352 +0x160
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc00033f960, 0xc0004afcc0, 0xc0004afd10, 0xc0000fe700, 0x18, 0x7efe66e11000)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc0003216c0, 0xc00048ce50, 0xc00048d090, 0x0, 0x0)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-tfe/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: reflect.Value.call(0xc0002e69c0, 0xc00000d860, 0x13, 0xed6f4d, 0x4, 0xc000095f18, 0x3, 0x3, 0xc0001896c0, 0x0, ...)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/goenv/versions/1.11.4/src/reflect/value.go:447 +0x454
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: reflect.Value.Call(0xc0002e69c0, 0xc00000d860, 0x13, 0xc00049e718, 0x3, 0x3, 0xc000189fc0, 0xc0000320a0, 0x722574)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/goenv/versions/1.11.4/src/reflect/value.go:308 +0xa4
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: net/rpc.(*service).call(0xc000345d80, 0xc000319360, 0xc00034e278, 0xc00034e290, 0xc0000c1e80, 0xc0003217a0, 0xd29920, 0xc00048ce50, 0x16, 0xd29960, ...)
2019-02-14T22:22:00.547Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/goenv/versions/1.11.4/src/net/rpc/server.go:384 +0x14e
2019-02-14T22:22:00.548Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4: created by net/rpc.(*Server).ServeCodec
2019-02-14T22:22:00.548Z [DEBUG] plugin.terraform-provider-tfe_v0.7.0_x4:       /opt/goenv/versions/1.11.4/src/net/rpc/server.go:481 +0x47e
2019-02-14T22:22:00.549Z [DEBUG] plugin: plugin process exited: path=/terraform/.terraform/plugins/linux_amd64/terraform-provider-tfe_v0.7.0_x4
2019/02/14 22:22:00 [ERROR] root: eval: *terraform.EvalRefresh, err: tfe_workspace.stack: unexpected EOF
2019/02/14 22:22:00 [ERROR] root: eval: *terraform.EvalSequence, err: tfe_workspace.stack: unexpected EOF
2019/02/14 22:22:00 [TRACE] [walkRefresh] Exiting eval tree: tfe_workspace.stack
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.region"
2019/02/14 22:22:00 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.tfe_team.pso: unexpected EOF
2019/02/14 22:22:00 [ERROR] root: eval: *terraform.EvalSequence, err: data.tfe_team.pso: unexpected EOF
2019/02/14 22:22:00 [TRACE] [walkRefresh] Exiting eval tree: data.tfe_team.pso
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_team_access.pso_read"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.credentials"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.secret_foo"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.secret_baz"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.cluster_name"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_team_access.phoogle_write"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.secret_bar"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.project"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "tfe_variable.workspace_owner"
2019/02/14 22:22:00 [TRACE] dag/walk: upstream errored, not walking "provider.tfe (close)"
2019/02/14 22:22:00 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-02-14T22:22:00.552Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,9 @@
-## 0.8.0 (Unreleased)
+## 0.7.1 (Unreleased)
+
+BUG FIXES:
+
+* r/tfe_workspace: Add a check when migrating `vcs_repo` from a set to a list [GH-64]
+
 ## 0.7.0 (February 14, 2019)
 
 ENHANCEMENTS:

--- a/tfe/resource_tfe_workspace.go
+++ b/tfe/resource_tfe_workspace.go
@@ -246,11 +246,11 @@ func resourceTFEWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
 
 		// Get and assert the VCS repo configuration block.
 		if v, ok := d.GetOk("vcs_repo"); ok {
-			vcsRepo := v.([]interface{})[0].(map[string]interface{})
-
-			// Only set the branch if one is configured.
-			if branch, ok := vcsRepo["branch"].(string); ok && branch != "" {
-				vcsConfig["branch"] = workspace.VCSRepo.Branch
+			if vcsRepo, ok := v.([]interface{})[0].(map[string]interface{}); ok {
+				// Only set the branch if one is configured.
+				if branch, ok := vcsRepo["branch"].(string); ok && branch != "" {
+					vcsConfig["branch"] = workspace.VCSRepo.Branch
+				}
 			}
 		}
 

